### ISSUE TITLE
Loading large chat log freezes game when loading world - 211.6.7

### ISSUE THREAD

**Issue by ARedstoneMan** [mrbuilder1961/WheresMyChatHistory#191]
### Description
Game freezes up when loading into a world or server on "Loading Terrain" screen and forces the user to force quit the game. When chatlog.json is deleted or renamed, game functions as normal (will link problematic chat log below)

### How to reproduce
1. Use given chat log
2. Load game
3. Attempt to join any server or world

### Expected result
When you attempt to join a world or server, it should load you in without freezing indefinitely. Also, something specific in the chat log should not prevent loading into a world.

### Log file link
https://mclo.gs/pFHVqEG (censored sensitive information)

### Specs and Details
Required information:
- Windows 11 Pro
- 210.6.5.1
- 0.15.11, 0.100.7+1.21, 3.5.0+1.21 (Fabric, Fabric API, YACL respectively)
- Mod Menu: 11.0.1

Optional but suggested information:
```json 
{
  "time": true,
  "timeDate": "HH:mm:ss",
  "timeFormat": "[$]",
  "timeColor": 16733695,
  "hover": true,
  "hoverDate": "MM/dd/yyyy",
  "hoverFormat": "$",
  "hoverColor": 16777215,
  "counter": true,
  "counterFormat": "\u00268(\u00267x\u0026r$\u00268)",
  "counterColor": 16777045,
  "counterCheckStyle": false,
  "counterCompact": false,
  "counterCompactDistance": 0,
  "boundary": true,
  "boundaryFormat": "\u00268[\u0026r$\u00268]",
  "boundaryColor": 5636095,
  "chatlog": true,
  "chatlogSaveInterval": 0,
  "chatHidePacket": true,
  "chatWidth": 0,
  "chatMaxMessages": 16384,
  "chatNameFormat": "\u003c$\u003e",
  "chatNameColor": 16777215,
  "shiftChat": 10,
  "messageDrafting": false,
  "onlyInvasiveDrafting": false,
  "searchDrafting": true,
  "hideSearchButton": false,
  "vanillaClearing": false,
  "copyColor": 5636095,
  "copyReplyFormat": "/msg $ "
}
```
- See attatched file below (e.json) for chatlog
[e.json](https://github.com/user-attachments/files/16433789/e.json)

**Comment by mrbuilder1961** [comment]
it seems like chatpatches didn't even load the chatlog completely, so i wonder if it was just taking a really long time to process all the data, although this is a concerning issue. the log shows 35 history messages and 5195 received messages

**Comment by mrbuilder1961** [comment]
i assume the issue will persist, but could you try loading the chatlog with [this dev version](https://cdn.discordapp.com/attachments/1077296221464559737/1267332482987196538/chatpatches-210.6.6-dev3.jar?ex=66afa719&is=66ae5599&hm=1cdbde883c1ec8ae1d0844b8173b38d79c3ffa9d4bd058f773161dd45d6dada5&) and lmk the results?

**Comment by ARedstoneMan** [comment]
Issue persists.

**Comment by nicko302** [comment]
i'm facing this issue too. an obvious fix is to disable the chat log, which does work in preventing the issue but ofc if you want the chat log (which i do) then it isnt a permanent fix.

**Comment by ARedstoneMan** [comment]
I may be wrong, but the way I understand it, a potential fix could be to clear old messages in the log after too many have been logged, to keep the file size low.

**Comment by mrbuilder1961** [comment]
> I may be wrong, but the way I understand it, a potential fix could be to clear old messages in the log after too many have been logged, to keep the file size low.

the message limit is set by the max message config option. so if you have a large value it will increase the footprint and time to load. that's not to say softlocking the game is intended, however.

**Comment by SirPotato42** [comment]
I also have this issue

**Comment by ttrafford7** [comment]
confirm this too

**Comment by DrKasango** [comment]
Tried the latest build, the issue persists.
https://mclo.gs/0olfsQv

Btw, tried to reduce the max message config option to 5000, still freezed on loading world.
Sorry for the confusion with the issue #180.

**Comment by mrbuilder1961** [comment]
No worries, thanks for re-uploading it over here. if possible, could you try to load the chat log again and just wait it out until the game either crashes or it loads? Then when you send the log, chat patches will note how long it took and I can use that for future debugging!

**Comment by DrKasango** [comment]
Last 10 times or so it was just loading terrain with neither crashing the game or letting me in for more than 10 minutes, while no new lines appeared in the latest.log file, so it ends exactly the same when I manually force the application to close. 
I can try again, of course, but I am not sure such a log will be of any help.

**Comment by mrbuilder1961** [comment]
okay i understand, that should be enough on its own. i am wondering is this on every server or just specific ones? and does this happen on single player at all?

additionally if you could upload your chat log file to pastebin or something that would be great and i can try to reproduce your scenario.

**Comment by DrKasango** [comment]
The same for other servers and for singleplayer as well.
Singleplayer log: 
https://mclo.gs/uiYdjg5
https://pastebin.com/WjxxjgCr

**Comment by DrKasango** [comment]
Tried 211.6.7 just in case, issue persists. 
https://mclo.gs/KZIZXgt

**Comment by RomanPro100** [comment]
Same issue, this was in 210.6.5 too and I think some versions before that

**Comment by RomanPro100** [comment]
Same issue, world loading takes around 2 MINUTES, both when joining Singleplayer and Multiplayer

**Comment by mrbuilder1961** [comment]
has anyone experienced this issue without any other mods installed, minus dependencies?

**Comment by DrKasango** [comment]
Here is the log without any other mods installed (except fabric api and kotlin):
https://mclo.gs/3mKirK6

**Comment by DrDerp04** [comment]
I also have this issue. are there any fixes for it?

**Comment by SirPotato42** [comment]
> I also have this issue. are there any fixes for it?

I second this

**Comment by mrbuilder1961** [comment]
here's a dev build you all can try and then report back with info and logs:
~~https://cdn.discordapp.com/attachments/1077296221464559737/1284343152882221188/chatpatches-211.6.7-dev1.jar?ex=66e64945&is=66e4f7c5&hm=bb1f53717e40931b38668616ecad78f18d12163fd37de644e719fdeabaa3e6d8&~~

**Comment by DrKasango** [comment]
Haven't noticed any changes:
https://mclo.gs/Bfe2EEr

**Comment by mrbuilder1961** [comment]
whoops, forgot some very obvious stuff. please try this one instead: https://cdn.discordapp.com/attachments/1077296221464559737/1284648566890958932/chatpatches-211.6.7-dev2.jar?ex=66e765b5&is=66e61435&hm=a757c95069aad13d655ba89e03778c3f99ebfaf367085b1c424c65709118abb8&

**Comment by DrKasango** [comment]
Seems to be fixed, thanks a lot!

**Comment by RomanPro100** [comment]
Seems to be fixed for me too, still a tiny bit (0.5-1s) slower than without the mod

**Comment by SirPotato42** [comment]
> whoops, forgot some very obvious stuff. please try this one instead: https://cdn.discordapp.com/attachments/1077296221464559737/1284648566890958932/chatpatches-211.6.7-dev2.jar?ex=66e765b5&is=66e61435&hm=a757c95069aad13d655ba89e03778c3f99ebfaf367085b1c424c65709118abb8&

![image](https://github.com/user-attachments/assets/3f0a6803-0193-4932-8206-2d62c74b095a)
link is broken

**Comment by mrbuilder1961** [comment]
try the build here, should be minutely faster https://github.com/mrbuilder1961/ChatPatches/actions/runs/10875600736

**Comment by DrDerp04** [comment]
Im having no problems with that build, thank you <3

**Comment by mrbuilder1961** [comment]
fixed in 211.6.8!
